src/js/components/modals/UpdateConfigModalContents.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+
+import AdvancedConfig from '../AdvancedConfig';
+import CosmosErrorMessage from '../CosmosErrorMessage';
+import CosmosPackagesStore from '../../stores/CosmosPackagesStore';
+import Service from '../../structs/Service';
+import ServicePlan from '../../structs/ServicePlan';
+import UniversePackage from '../../structs/UniversePackage';
+
+const METHODS_TO_BIND = [
+  'getConfigSubmit',
+  'handleAdvancedFormChange',
+  'handleConfigSave'
+];
+
+class UpdateConfigModal extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      hasFormErrors: false
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleAdvancedFormChange(formData) {
+    this.setState({hasFormErrors: !formData.isValidated});
+  }
+
+  handleConfigSave() {
+    let serviceMetadata = this.props.service.getMetadata();
+    let {name, version} = serviceMetadata;
+
+    let {isValidated, model} = this.triggerConfigSubmit();
+
+    if (isValidated) {
+      // TODO: Create the proper action for updating the configuration,
+      // install is not the correct action.
+      // https://mesosphere.atlassian.net/browse/DCOS-6729
+      CosmosPackagesStore.installPackage(name, version, model);
+      this.props.setPendingRequest(true);
+    }
+  }
+
+  getConfigSubmit(triggerSubmit) {
+    this.triggerConfigSubmit = triggerSubmit;
+  }
+
+  getUpdateErrorContent() {
+    let {updateError} = this.props;
+
+    return (
+      <CosmosErrorMessage
+        className="text-error-state text-overflow-break-word text-align-center"
+        error={updateError}
+        wrapperClass="modal-content-inner container container-pod
+          container-pod-short horizontal-center" />
+    );
+  }
+
+  getMarathonConfigurationModalContents() {
+    // TODO: The marathon configuration modal content needs to be returned.
+    // https://mesosphere.atlassian.net/browse/DCOS-6639
+    return null;
+  }
+
+  getModalContent(servicePackage) {
+    if (servicePackage instanceof UniversePackage) {
+      return this.getPackageConfigurationModalContents(servicePackage);
+    }
+
+    return modalContent = this.getMarathonConfigurationModalContents(servicePackage);
```

Comment:
I think this is what you meant:
`return this.getMarathonConfigurationModalContents(servicePackage);`